lib/ex_uid2/encryption.ex

```diff
@@ -0,0 +1,41 @@
+defmodule ExUid2.Encryption do
+  alias ExUid2.Encryption.EncryptedToken
+  alias ExUid2.Encryption.Identity
+  alias ExUid2.Encryption.MasterPayload
+  alias ExUid2.Uid2
+  alias ExUid2.Keyring
+
+  @spec decrypt_v2_token(binary(), Keyring.t(), integer()) :: Uid2.t() | {:error, any()}
+  def decrypt_v2_token(
```

Comment:
Is v2 meant to only decrypt to UID2? Just wondering how do you tell which version of the decryption you're supposed to use.

Reply:
Yes, V2 is meant to decrypt only UID2. Technically, we can branch on which version to decrypt by looking at the first 2 bytes of the decoded payload.

If the first byte is `2`, then it's a V2 token. Otherwise, if the second byte holds `112`, it's a V3 token. If it holds `128`, it's a V4 token. I don't know how to identify V1 tokens.

In our case, we're currently only interested in decrypting V2 tokens, so a basic check is made [here](https://github.com/market-ops/ex_uid2/blob/dev/lib/ex_uid2/encryption/encrypted_token.ex#L22) and if it's not a V2 token, it will return `{:error, :invalid_v2_token}`. If we ever need to branch on different token versions, it'll be easy to replace the call to `Encryption.decrypt_v2_token/3` with something that parses and branches first, but for now since we only need to support V2 that seemed unnecessary.

Reply:
Alrighty, thanks for the clarification.